ray-operator/controllers/ray/utils/util.go

```diff
@@ -346,6 +346,9 @@ func PodNotMatchingTemplate(pod corev1.Pod, template corev1.PodTemplateSpec) boo
 					// resource entries do not match
 					return true
 				}
+				if compareVolumeMounts(container1.VolumeMounts, container2.VolumeMounts) {
```

Comment:
I think we should `return true` only when the volume mounts are different.
```suggestion
				if compareVolumeMounts(container1.VolumeMounts, container2.VolumeMounts) {
```

Reply:
Function no longer exists.